### PR TITLE
feat(rust): use default node when -at argument is none

### LINKED ISSUE #5332: Use default node when none is specified in the `--at` argument of the `policy create` command
### Current behavior

This command has no default value for the `--at` argument.

```rust
#[arg(value_name = "NODE", long)]
at: String,
```

### Desired behavior

Modify the `--at` argument of this command to add the default behavior described below.

By convention, all commands acting on a node should have an `--at` argument that defaults to the default node when no value is passed. This argument is defined as follows ([example here](https://github.com/build-trust/ockam/blob/c4e94dbd0d633896f250d15b2227b1d17993174d/implementations/rust/ockam/ockam_command/src/secure_channel/list.rs#L40)):

```rust
#[arg(value_name = "NODE_NAME", long)]  
at: Option<String>,
```

The argument is parsed in the command logic as follows ([example here](https://github.com/build-trust/ockam/blob/c4e94dbd0d633896f250d15b2227b1d17993174d/implementations/rust/ockam/ockam_command/src/secure_channel/list.rs#L91-L92)):

```rust
let at = get_node_name(&opts.state, &cmd.at);  
let node_name = parse_node_name(&at)?;
```

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

### LINKED ISSUE #5333: Use default node when none is specified in the `--at` argument of the `policy delete` command
### Current behavior

This command has no default value for the `--at` argument.

```rust
#[arg(value_name = "NODE", long)]
at: String,
```

### Desired behavior

Modify the `--at` argument of this command to add the default behavior described below.

By convention, all commands acting on a node should have an `--at` argument that defaults to the default node when no value is passed. This argument is defined as follows ([example here](https://github.com/build-trust/ockam/blob/c4e94dbd0d633896f250d15b2227b1d17993174d/implementations/rust/ockam/ockam_command/src/secure_channel/list.rs#L40)):

```rust
#[arg(value_name = "NODE_NAME", long)]  
at: Option<String>,
```

The argument is parsed in the command logic as follows ([example here](https://github.com/build-trust/ockam/blob/c4e94dbd0d633896f250d15b2227b1d17993174d/implementations/rust/ockam/ockam_command/src/secure_channel/list.rs#L91-L92)):

```rust
let at = get_node_name(&opts.state, &cmd.at);  
let node_name = parse_node_name(&at)?;
```

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

### LINKED ISSUE #5334: Use default node when none is specified in the `--at` argument of the `policy list` command
### Current behavior

This command has no default value for the `--at` argument.

```rust
#[arg(value_name = "NODE", long)]
at: String,
```

### Desired behavior

Modify the `--at` argument of this command to add the default behavior described below.

By convention, all commands acting on a node should have an `--at` argument that defaults to the default node when no value is passed. This argument is defined as follows ([example here](https://github.com/build-trust/ockam/blob/c4e94dbd0d633896f250d15b2227b1d17993174d/implementations/rust/ockam/ockam_command/src/secure_channel/list.rs#L40)):

```rust
#[arg(value_name = "NODE_NAME", long)]  
at: Option<String>,
```

The argument is parsed in the command logic as follows ([example here](https://github.com/build-trust/ockam/blob/c4e94dbd0d633896f250d15b2227b1d17993174d/implementations/rust/ockam/ockam_command/src/secure_channel/list.rs#L91-L92)):

```rust
let at = get_node_name(&opts.state, &cmd.at);  
let node_name = parse_node_name(&at)?;
```

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

### LINKED ISSUE #5335: Use default node when none is specified in the `--at` argument of the `secure-channel delete` command
### Current behavior

This command has no default value for the `--at` argument.

```rust
#[arg(value_name = "NODE", long)]
at: String,
```

### Desired behavior

Modify the `--at` argument of this command to add the default behavior described below.

By convention, all commands acting on a node should have an `--at` argument that defaults to the default node when no value is passed. This argument is defined as follows ([example here](https://github.com/build-trust/ockam/blob/c4e94dbd0d633896f250d15b2227b1d17993174d/implementations/rust/ockam/ockam_command/src/secure_channel/list.rs#L40)):

```rust
#[arg(value_name = "NODE_NAME", long)]  
at: Option<String>,
```

The argument is parsed in the command logic as follows ([example here](https://github.com/build-trust/ockam/blob/c4e94dbd0d633896f250d15b2227b1d17993174d/implementations/rust/ockam/ockam_command/src/secure_channel/list.rs#L91-L92)):

```rust
let at = get_node_name(&opts.state, &cmd.at);  
let node_name = parse_node_name(&at)?;
```

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

### PR DESCRIPTION
**Pull request by aleksandarskrbic**
## Proposed changes

Use default node when -at argument is none of the next commands:
`secure-channel delete`
`policy delete`
`policy list`
`policy create`

https://github.com/build-trust/ockam/issues/5332
https://github.com/build-trust/ockam/issues/5333
https://github.com/build-trust/ockam/issues/5334
https://github.com/build-trust/ockam/issues/5335

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

### CODE REVIEW

**Comment by mrinalwadhwa** [comment]
@aleksandarskrbic Thank you for spending time on this, while we review, please:

- [x] Accept the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding your Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will make the commit as verified.

**Comment by mrinalwadhwa** [comment]
Ah, never mind, I remembered that you already accepted :)
https://github.com/build-trust/ockam-contributors/pull/245

**Review by adrianbenavides — CHANGES_REQUESTED** [review]
Hey @aleksandarskrbic thanks for the PR!

The changes you propose add a naive fallback to a "default" node, but we also want to make this fallback more robust and improve the UX of these commands:
- Note that the "default" node it's not necessarily the system's default node. The default node can be retrieved using `opts.state.nodes.default()`
- Adding a fallback in the command handler is still not enough from the UX perspective, as the user is still forced to pass the `--at` argument. That is, the argument doesn't have a default value, you are just adding it after the argument was parsed.

Make sure to re-read the issues' description: https://github.com/build-trust/ockam/issues/5333. You can find there how to make the argument optional and how to fallback to the default node without depening on a hardcoded string.

And, of course, feel free to ask any question!

File: implementations/rust/ockam/ockam_command/src/policy/delete.rs [hunk 1]
```diff
@@ -43,21 +44,22 @@ async fn run_impl(
...
         rpc.is_ok()?;
 
         opts.terminal
             .stdout()
-            .plain(fmt_ok!("Policy with name '{}' has been deleted", &cmd.at))
-            .machine(&cmd.at)
-            .json(serde_json::json!({ "policy": { "at": &cmd.at } }))
+            .plain(fmt_ok!("Policy with name '{}' has been deleted", &at))
+            .machine(&at)
+            .json(serde_json::json!({ "policy": { "at": &at} }))
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/policy/delete.rs]: Here you should be using the `node_name` variable, as the `at` variable could contain a `MultiAddr` formatted string (e.g. `/node/n1`, when we actually want to display just the `n1` part).

**Review by adrianbenavides — COMMENTED** [review]
Looks perfect now, just one small fix on the output of one of the commands and will be ready to merge!

File: implementations/rust/ockam/ockam_command/src/policy/delete.rs (hunk 1 above)
Comment (aleksandarskrbic) [reply implementations/rust/ockam/ockam_command/src/policy/delete.rs]: @adrianbenavides Done! :)

**Review by adrianbenavides — APPROVED** [review]
Nice, thank you so much @aleksandarskrbic for another great PR! 🚀

**Comment by aleksandarskrbic** [comment]
> Nice, thank you so much @aleksandarskrbic for another great PR! 🚀

Thank you for helping me! :)

**Review by etorreborre — APPROVED** [review]

**Pull request merged**